packages/data/addon/services/keg.js

```diff
@@ -77,21 +82,22 @@ export default class KegService extends Service {
     const factory = this._getFactoryForType(options.modelFactory || type);
     const recordKeg = this._getRecordKegForType(type);
     const idIndex = this._getIdIndexForType(type);
+    const namespace = options.namespace || this.defaultNamespace;
     const identifierField = factory.identifierField || KegService.identifierField;
     const owner = getOwner(this);
 
     const returnedRecords = A();
     for (let i = 0; i < rawRecords.length; i++) {
       const id = rawRecords[i][identifierField];
-      const existingRecord = this.getById(type, id);
+      const existingRecord = this.getById(type, id, namespace);
 
       if (existingRecord) {
         setProperties(existingRecord, rawRecords[i]);
         returnedRecords.pushObject(existingRecord);
       } else {
         let newRecord = factory.create(Object.assign({}, owner.ownerInjection(), rawRecords[i]));
 
-        idIndex[id] = newRecord;
+        idIndex[`${namespace}.${id}`] = newRecord;
```

Comment:
would it be worth it to make keg have `namespace > type > id`?

Reply:
I think that would be a big refactor, not sure if it's worth it right now.